db/db_impl.cc

```diff
@@ -1153,6 +1154,28 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& log_numbers,
       }
       WriteBatchInternal::SetContents(&batch, record);
 
+#ifndef ROCKSDB_LITE
+      if (db_options_.wal_filter != nullptr) {
+        WALFilter::WALProcessingOption walProcessingOption =
+          db_options_.wal_filter->LogRecord(batch);
+
+        switch (walProcessingOption) {
+        case  WALFilter::WALProcessingOption::kContinueProcessing:
+          //do nothing, proceeed normally
+          break;
+        case WALFilter::WALProcessingOption::kIgnoreCurrentRecord:
+          //skip current record
+          continue;
+        case WALFilter::WALProcessingOption::kStopReplay:
+          //skip current record and stop replay
+          continue_replay_log = false;
+          continue;
+        default:
+          assert(false); //unhandled case
+        }
+      }
```

Comment:
If application itself commits a batch with LogData what would happen? Would there be multiple LogData in one batch here? If that is true, it might be good to break it down. How can we get each atomic batch from WriteBatch here?